lib/src/logic.dart

```diff
@@ -12,9 +12,10 @@ import 'dart:async';
 
 import 'package:collection/collection.dart';
 import 'package:meta/meta.dart';
+
 import 'package:rohd/rohd.dart';
+import 'package:rohd/src/types/synchronous_propagator.dart';
```

Comment:
I'm rethinking calling this folder "types", sorry.  I think it may confuse users since it sounds like these might be types of other things such as hardware constructs.  Maybe let's keep synchronous_propagator in utilities/ and we can rename this folder to something like "collections"?

Reply:
Yup, sounds good.